x-pack/plugins/apm/server/routes/services/get_services_detailed_statistics/get_service_aggregated_transaction_detailed_statistics.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { keyBy } from 'lodash';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import {
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+  TRANSACTION_DURATION_SUMMARY,
+  TRANSACTION_FAILURE_COUNT,
+  TRANSACTION_SUCCESS_COUNT,
+} from '../../../../common/elasticsearch_fieldnames';
+import { withApmSpan } from '../../../utils/with_apm_span';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { environmentQuery } from '../../../../common/utils/environment_query';
+import { getOffsetInMs } from '../../../../common/utils/get_offset_in_ms';
+import { calculateThroughputWithRange } from '../../../lib/helpers/calculate_throughput';
+import { getBucketSizeForAggregatedTransactions } from '../../../lib/helpers/get_bucket_size_for_aggregated_transactions';
+import { Setup } from '../../../lib/helpers/setup_request';
+import { calculateFailedTransactionRateFromServiceMetrics } from '../../../lib/helpers/transaction_error_rate';
+import { RandomSampler } from '../../../lib/helpers/get_random_sampler';
+
+export async function getServiceAggregatedTransactionDetailedStats({
+  serviceNames,
+  environment,
+  kuery,
+  setup,
+  searchAggregatedServiceMetrics,
+  offset,
+  start,
+  end,
+  randomSampler,
+}: {
+  serviceNames: string[];
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  searchAggregatedServiceMetrics: boolean;
+  offset?: string;
+  start: number;
+  end: number;
+  randomSampler: RandomSampler;
+}) {
+  const { apmEventClient } = setup;
+  const { offsetInMs, startWithOffset, endWithOffset } = getOffsetInMs({
+    start,
+    end,
+    offset,
+  });
+
+  const metrics = {
+    avg_duration: {
+      avg: {
+        field: TRANSACTION_DURATION_SUMMARY,
+      },
+    },
+    total_doc: {
+      value_count: {
+        field: TRANSACTION_DURATION_SUMMARY,
+      },
+    },
+    failure_count: {
+      sum: {
+        field: TRANSACTION_FAILURE_COUNT,
+      },
+    },
+    success_count: {
+      sum: {
+        field: TRANSACTION_SUCCESS_COUNT,
+      },
+    },
+  };
+
+  const response = await apmEventClient.search(
+    'get_service_aggregated_transaction_detail_stats',
+    {
+      apm: {
+        events: [ProcessorEvent.metric],
+      },
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              { terms: { [SERVICE_NAME]: serviceNames } },
+              ...[{ exists: { field: TRANSACTION_DURATION_SUMMARY } }],
```

Comment:
Can we use a term query on metricset.name rather than an exists query?

Reply:
sure,  is `term` faster?

Reply:
It very well might be, but it's also more explicit. It's basically the difference between "I want this kind of document" and "I want a document that has this attribute".

Reply:
Makes sense. I updated it everywhere https://github.com/elastic/kibana/pull/140868/commits/5a7475f6f2f50be36c60bbb8ef5a0fb98b3cb3be